### PR TITLE
Add e2e test for plugin archetype

### PR DESCRIPTION
**Pull request by Flanker32**
Add e2e test for plugin archetype, for issue #78

### CODE REVIEW

**Comment by Flanker32** [comment]
@pragnagopa  Please help take a look when you have time.

File: e2etest.ps1 [hunk 1]
```diff
@@ -0,0 +1,30 @@
+$base = pwd;
+# Download Functions Core Tools
+Remove-Item -Force "$base\Azure.Functions.Cli.zip" -ErrorAction Ignore
+Remove-Item -Recurse -Force "$base\Azure.Functions.Cli" -ErrorAction Ignore
+Write-Host "Downloading Functions Host...."
+$url = "https://functionsclibuilds.blob.core.windows.net/builds/2/latest/Azure.Functions.Cli.win-x64.zip"
+$output = "$base\Azure.Functions.Cli.zip"
```
Comment (SummerSun) [inline e2etest.ps1]: Define a variable for multiple usage.

File: e2etest.ps1 [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+$wc = New-Object System.Net.WebClient
+$wc.DownloadFile($url, $output)
```
Comment (SummerSun) [inline e2etest.ps1]: No need to define a $wc

File: e2etest.ps1 [hunk 3]
```diff
@@ -0,0 +1,30 @@
+$base = pwd;
+# Download Functions Core Tools
+Remove-Item -Force "$base\Azure.Functions.Cli.zip" -ErrorAction Ignore
+Remove-Item -Recurse -Force "$base\Azure.Functions.Cli" -ErrorAction Ignore
+Write-Host "Downloading Functions Host...."
+$url = "https://functionsclibuilds.blob.core.windows.net/builds/2/latest/Azure.Functions.Cli.win-x64.zip"
```
Comment (SummerSun) [inline e2etest.ps1]: Define in environment variable

File: e2etest.ps1 [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+$output = "$base\Azure.Functions.Cli.zip"
+$wc = New-Object System.Net.WebClient
+$wc.DownloadFile($url, $output)
+Expand-Archive "$base\Azure.Functions.Cli.zip" -DestinationPath "$base\Azure.Functions.Cli"
+$Env:Path = $Env:Path+";$base\Azure.Functions.Cli"
+# Install maven plguin archetype
+$archetypePom = Get-Content ".\azure-functions-archetype\pom.xml" -Raw
+$archetypePom -match "<version>(.*)</version>"
```
Comment (SummerSun) [inline e2etest.ps1]: Take a look at the Powershell Pipeline

File: e2etest.ps1 [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+$atchetypeVersion = $matches[1]
```
Comment (SummerSun) [inline e2etest.ps1]: Add comment or use a function to get the value

File: e2etest.ps1 [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+mvn clean install
+# Generate function project through archetype
+Remove-Item -Recurse -Force ".\e2etestproject" -ErrorAction Ignore
```
Comment (SummerSun) [inline e2etest.ps1]: Remove if the directory exists

File: e2etest.ps1 [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+mkdir e2etestproject
+cd e2etestproject
```
Comment (SummerSun) [inline e2etest.ps1]: better naming, avoid use the same name as the subfolder name

File: e2etest.ps1 [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    }
+}
+
+Function RemoveFolderIfExist($folderName){
+    if(Test-Path $folderName){
+        Remove-Item -Recurse -Force $folderName
+    }
+}
+
+Function DownloadFileFromUrl($url,$destination){
```
Comment (SummerSun) [inline e2etest.ps1]: Format, add a whitespace between parameters

File: e2etest.ps1 [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+DownloadFileFromUrl $Env:FUNCTIONCLI_URL $functionCLIZipPath
+Expand-Archive $functionCLIZipPath -DestinationPath $functionCLIPath
+$Env:Path = $Env:Path+";$functionCLIPath"
+
+# Install maven plguin archetype
+mvn clean install
+
+# Generate function project through archetype
+$testProjectBaseFolder = ".\testprojects"
+# Get the version of archetype and set archetypeVersion or maven will use the repository version to create project
```
Comment (SummerSun) [inline e2etest.ps1]: This comment is a little confusing. 
I think just `Get the version of archetype and set archetypeVersion` is enough.

File: e2etest.ps1 (hunk 8 above)
Comment (Flanker32) [reply e2etest.ps1]: Solved

File: e2etest.ps1 (hunk 9 above)
Comment (Flanker32) [reply e2etest.ps1]: Done~

**Pull request merged**